src/deepsparse/transformers/pipelines/nli_text_classification.py

```diff
@@ -0,0 +1,268 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# postprocessing adapted from huggingface/transformers
+
+# Copyright 2021 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Helper functions and pydantic models for zero-shot text classification task
+with nli models
+"""
+
+
+from concurrent.futures import wait
```

Comment:
given the purpose of these helper functions is for shared functionality across potential future pipelines, and these helper functions assume properties of the given pipelines such as `_threadpool`, would it make sense to refactor this into a parent nli pipeline class that handles the processing as well as creation of the threadpool, config, etc?

Reply:
I think the vision was to allow for users to choose between nli and future implementations of zero shot via the model_scheme constructor arg. I'm not aware of any ways to dynamically inherit from a parent class at construction time.

If anything it seems like the nli class should inherit from a zero shot pipeline class, but I believe that implementation will break once new child classes are added, since they can't all be registered under the same "zero_shot_text_classification" task.

After investigating a bit more, perhaps we can override the `__new__` method, which will allow the construction of the `ZeroShotTextClassificationPipeline` to dynamically decide to return a `MnliTextClassificationPipeline`. The mnli pipeline will then inherit from the zero shot pipeline and override its own methods.

Reply:
```
class ZeroShotTextClassificationPipeline():
    def __new__(cls, model_scheme):
        print(f"ZeroShotTextClassificationPipeline __new__")

        if model_scheme == "mnli":
            return MnliTextClassificationPipeline()
        elif model_scheme == "mlm":
            return MlmTextClassificationPipeline()
        else:
            raise ValueError()

class MlmTextClassificationPipeline(TransformersPipeline):
    # cannot inherit from ZeroShotTextClassificationPipeline
    pass

class MnliTextClassificationPipeline(TransformersPipeline):
    # cannot inherit from ZeroShotTextClassificationPipeline
    pass
```